File: OCDETF.StrikeForce.Web.API/ClientApp/src/app/models/quarterly-report.ts
import { QuarterlyActivity } from "./quarterly-activity";
import { RecentDevelopment } from "./recent-development";
import { Staffing } from "./staffing";

export class QuarterlyReport {
  ID: string;
  Name: string;
  StrikeForceName: string;
  StrikeForceID: string;
  FiscalYear: string;
  Quarter: string;
  OperationsBegin: string;
  MOUDate: string;
  HistoryNotes: string;
  Mission?: string;
  Structure?: string;
  OfficeLocations: string;  
  Challenges: string;
  HeadsUp: string;
  SeizureNotes: string;
  SpecificNotes: string;
  IsActive: boolean;
  Status: string;

  CreatedUserID: string;
  UpdatedUserID: string;
  DateCreated: Date;
  DateUpdated: Date;

  PartitionKey: string;
  RowKey: string;

  StaffingData: Staffing[];
  RequiredData: QuarterlyActivity[];
  OCDETFCases: QuarterlyActivity[];
  SeizuresData: QuarterlyActivity[];
  SpecificData: QuarterlyActivity[];
  RecentDevelopments: RecentDevelopment[];

  constructor() {
    this.ID = "";
    this.Name = "";
    this.StrikeForceName = "";
    this.StrikeForceID = "";
    this.FiscalYear = "";
    this.Quarter = "";
    this.OperationsBegin = "";
    this.MOUDate = "";
    this.HistoryNotes = "";
    this.OfficeLocations = "";
    this.StaffingData = [];
    this.RequiredData = [];
    this.OCDETFCases = [];

    this.SeizuresData = [];
    this.SpecificData = [];
    this.SeizureNotes = "";
    this.Challenges = "";
    this.RecentDevelopments = [];
    this.HeadsUp = "";
    this.Status = "";
    this.SpecificNotes = "";
    this.CreatedUserID = "";
    this.UpdatedUserID = "";
    this.DateCreated = new Date();
    this.DateUpdated = new Date();
    this.PartitionKey = "";
    this.RowKey = "";
    this.IsActive = false;
  }
}

File: OCDETF.StrikeForce.Business.Library/Models/QuarterlyReport.cs
using Azure;
using Azure.Data.Tables;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;
using System.Text;
using Newtonsoft.Json;

namespace OCDETF.StrikeForce.Business.Library.Models
{
    public class QuarterlyReport :ITableEntity
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string ID { get; set; }
        public string Name { get; set; }
        public string StrikeForceID { get; set; }
        public string StrikeForceName { get; set; }        
        public int Quarter { get; set; }
        public int FiscalYear { get; set; }
        public string OperationsBegin { get; set; }
        public string MOUDate { get; set; }
        public string HistoryNotes { get; set; }
        public string Mission { get; set; }
        public string Structure { get; set; }
        public string OfficeLocations { get; set; }
        public string Challenges { get; set; }
        public string HeadsUp { get; set; }
        public string SeizureNotes { get; set; }
        public string SpecificNotes { get; set; }
        public bool IsActive { get; set; }
        public string Status { get; set; }

        public string CreatedUserID { get; set; }
        public string UpdatedUserID { get; set; }
        public DateTime DateUpdated { get; set; }
        public DateTime DateCreated { get; set; }

        public string PartitionKey { get; set; }
        public string RowKey { get; set; }
        public DateTimeOffset? Timestamp { get; set; }
        
        [BsonIgnore]
        [JsonIgnore]
        public ETag ETag { get; set; }
    }
}
